components/AvalancheForecastZoneMap.tsx

```diff
@@ -92,13 +102,112 @@ export const CARD_WHITESPACE = (1 - CARD_WIDTH) / 2; // proportion of overall wi
 export const CARD_SPACING = CARD_WHITESPACE / 2; // proportion of overall width that the spacing between two cards takes up
 export const CARD_MARGIN = CARD_SPACING / 2; // proportion of overall width that each card needs as a margin
 
+enum AnimatedDrawerState {
+  Hidden = 'Hidden',
+  Docked = 'Docked',
+  Visible = 'Visible',
+}
+
+class AnimatedDrawerController {
```

Comment:
using a traditional class here, since the animation state can't be used to drive React state